docs/codeql/codeql-language-guides/data-extensions-to-model-java-dependencies.rst

```diff
@@ -0,0 +1,295 @@
+.. _data-extensions-to-model-dependencies:
+
+Data extensions to model your Java/Kotlin dependencies
+======================================================
+
+You can use data extensions to model the methods and callables that control data flow in any framework or library. This is especially useful for custom frameworks or niche libraries, that are not supported by the standard CodeQL libraries.
+
+.. include:: ../reusables/kotlin-beta-note.rst
+
+.. include:: ../reusables/beta-note-model-packs-java.rst
+
+About this article
+------------------
+
+This article contains reference material about how data extensions interact with standard queries and the syntax used to define extensions.
+
+If you want to create your own data extensions, you should use the CodeQL model editor in the CodeQL extension for Visual Studio Code. The model editor automatically guides you through the process of defining models, displaying the properties you need to define and the options available. You can save the resulting models as data extensions and use them without worrying about the syntax.
```

Comment:
Data extensions are more general than the specific extensible predicates used for MaD modelling. We could imagine other uses cases than MaD for data extensions - we expect to use extensible predicates for threat modelling as well.

Reply:
Thanks for flagging this. I'll leave @jf205 to decide whether we should generalize this now, or in the future when document threat modeling ✨